source/material_model/rheology/diffusion_creep.cc

```diff
@@ -36,24 +37,44 @@ namespace aspect
       DiffusionCreep<dim>::DiffusionCreep ()
       {}
 
-
-
       template <int dim>
       double
       DiffusionCreep<dim>::compute_viscosity (const double pressure,
                                               const double temperature,
-                                              const unsigned int composition) const
-      {
+                                              const unsigned int composition,
+                                              const std::pair<std::vector<double>*, const std::vector<unsigned int>*> *gamma_inputs) const
```

Comment:
Hi Rene

I am working on the pull request row, but I find out that in gamma_inputs with the form of  'const std::pair<std::vector<double>*, const std::vector<unsigned int>*> *', the two vectors are not the same size. That is because the first one has the length of phases while the second one has the length of compositions. Thus, it can not be changed into the other form of it. Please let me know what you think of it.

Reply:
Hi Haoyuan, sorry for coming back late to your comment. If the two vectors have different lengths and content that is not directly associated with each other that suggest that they maybe should not be a combined parameter, but instead two separate ones (one called `n_phases_per_composition` and one `phase_function_values`), do you think that makes sense?